core/src/main/java/feign/FeignException.java

```diff
@@ -85,4 +124,100 @@ static FeignException errorExecuting(Request request, IOException cause) {
         cause,
         null);
   }
+
+  public static class BadRequest extends FeignException {
```

Comment:
I feel like all the 4xx ones should extend some sort of 4xx exception (say `FiengClientException`) and the same with 5xx errors (maybe `FeignServerException`).